pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -6607,9 +5860,9 @@ func TestReconcilePipeline_FinalTasks(t *testing.T) {
 					t.Errorf("Expected PipelineRun status to be failed, but was %v for %s",
 						reconciledRun.Status.GetCondition(apis.ConditionSucceeded), tt.name)
 				}
-				if d := cmp.Diff(reconciledRun.Status.TaskRuns, tt.expectedTaskRuns); d != "" {
-					t.Fatalf("Expected PipelineRunTaskRun status to match TaskRun(s) status, but got a mismatch for %s: %s", tt.name, d)
-				}
+				// if d := cmp.Diff(reconciledRun.Status.TaskRuns, tt.expectedTaskRuns); d != "" {
```

Comment:
Wondering if it is just safe here to remove the expected TaskRun, not sure if any checks needs to be added functionailty-wise?

Reply:
you could use the tekton client to query the status of the taskruns named in the pipelinerun.status.childreferences and assert that their statuses are as expected.